### PR TITLE
Add Censys collector

### PR DESCRIPTION
**Pull request by vortexau**
About Censys, from their about page.
"Censys is a search engine that enables researchers to ask questions about the hosts and networks that compose the Internet. Censys collects data on hosts and websites through daily ZMap and ZGrab scans of the IPv4 address space, in turn maintaining a database of how hosts and websites are configured. Researchers can interact with this data through a search interface, report builder, and SQL engine. Details on the Censys architecture are available in our research paper."

I and other researchers have found subdomains in Censys as part of TLS certs that would simply be impossible to find via bruteforcing.

I have developed a basic Censys integration for Aquatone - initially this only searches for subdomains in the TLS certificates Censys knows about, but can be expanded from here. 

Using the 'corp.yahoo.com' domain as an example, the following results were achieved:

aquatone v0.1.1 returns 12249 hosts:
aquatone-1 1

aquatone with Censys returns **12509** hosts:
aquatone-1 2

Censys found 524 hosts total, and **260** hosts not previously found in other services.

Censys requires two keys, a UserID and the API Key.

I also bumped the version number to v0.1.2 and added to the Changelog in my PR - please feel free to ignore if this does not suit.

### CODE REVIEW

**Comment by michenriksen** [comment]
Woah, this is pretty amazing - thank you very much!

Could you roll back your changes to the changelog and the gem version? I will coordinate the version bumping and changelog when I release the new version. I will make sure to credit you for this contribution. :)

**Comment by vortexau** [comment]
Done, changelog and version rolled back.

File: lib/aquatone/collectors/censys.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
+module Aquatone
+  module Collectors
+    class Censys < Aquatone::Collector
+      self.meta = {
+        :name         => "Censys",
+        :author       => "James McLean (@vortexau)",
+        :description  => "Uses the Censys API to find hostnames in TLS certificates",
+        :require_keys => ["censys","censysid"],
```
Comment (michenriksen) [inline lib/aquatone/collectors/censys.rb]: I'm sorry, just one more request: Could we change the key names to `censys_id` and `censys_secret` to be consistent with Censys' own wording and other aquatone collectors?

Thanks!

**Comment by vortexau** [comment]
have made the change, that definately makes it clearer.

**Comment by michenriksen** [comment]
Awesome, thank you!

**Pull request merged**